File: Python/8py 4/to_alternating_case.py
"""
|--------------------------------------------------------------------------
| altERnaTIng cAsE <=> ALTerNAtiNG CaSe
|--------------------------------------------------------------------------
|
"""

def to_alternating_case(string):
   return "".join([i.lower() if i.isupper() else i.upper() for i in list(string)])


def to_alternating_case(string):
   return string.swapcase()


def to_alternating_case(string):
   res = ''
   for c in string:
      res += c.upper() if c.islower() else c.lower()
   return res


print(to_alternating_case("hello WORLD"), "HELLO world")
print(to_alternating_case("HeLLo WoRLD"), "hEllO wOrld")
print(to_alternating_case("12345"), "12345")
print(to_alternating_case("1a2b3c4d5e"), "1A2B3C4D5E")

File: JavaScript/8js 4/toAlternatingCase.js
/*
|--------------------------------------------------------------------------
| altERnaTIng cAsE <=> ALTerNAtiNG CaSe
|--------------------------------------------------------------------------
|
*/

String.prototype.toAlternatingCase = function () {
   let res = "";
   this.split("").forEach((i) => {
      res += i == i.toLowerCase() ? i.toUpperCase() : i.toLowerCase();
   });
   return res;
};


String.prototype.toAlternatingCase = function () {
   return this.split("")
      .map((a) => (a == a.toUpperCase() ? a.toLowerCase() : a.toUpperCase()))
      .join("");
};


String.prototype.toAlternatingCase = function () {
   return this.replace(/[A-Za-z]/g, (x) => (x > "Z" ? x.toUpperCase() : x.toLowerCase()));
};


console.log("hello WORLD".toAlternatingCase(), "HELLO world");
console.log("HeLLo WoRLD".toAlternatingCase(), "hEllO wOrld");
console.log("12345".toAlternatingCase(), "12345");
console.log("1a2b3c4d5e".toAlternatingCase(), "1A2B3C4D5E");